src/store/modules/canvas.ts

```diff
@@ -61,50 +61,46 @@ const CanvasModule: Module<CanvasState, RootState> = {
     [CanvasMutation.SET_CANVAS_ELEMENT] (state: CanvasState, payload: CanvasElement[]) {
       state.canvasElements = [...payload]
     },
-    [CanvasMutation.HIDE_CANVAS_ELEMENT] (state: CanvasState, payload: HideCanvasElementInterface) {
-      if (payload.id) {
-        const foundElement = state.canvasElements.find((canvasElement: CanvasElement) => canvasElement.id === payload.id)
-        if (foundElement) {
-          foundElement.tracker = Tracker.REMOVAL
-          if (!payload.fromSocket) {
-            state.canvasElementsHistory.push({ ...foundElement })
-          }
-        }
+    [CanvasMutation.SET_CANVAS_ELEMENT_HISTORY] (state: CanvasState, payload: CanvasElementHistory[]) {
+      state.canvasElementsHistory = [...payload]
+    },
+    [CanvasMutation.HIDE_CANVAS_ELEMENT] (state: CanvasState, payload: CanvasElement) {
+      const foundElement = state.canvasElements.find((canvasElement: CanvasElement) => canvasElement.id === payload.id)
+      if (foundElement) {
+        foundElement.isVisible = false
       }
     },
-    [CanvasMutation.DELETE_CANVAS_ELEMENT] (state: CanvasState, payload: RootState) {
-      if (payload.id) {
-        const foundElement = state.canvasElements.find((canvasElement: CanvasElement) => canvasElement.id === payload.id)
-        if (foundElement) {
-          const index = state.canvasElements.indexOf(foundElement)
-          state.canvasElements.splice(index, 1)
-        }
+    [CanvasMutation.SHOW_CANVAS_ELEMENT] (state: CanvasState, payload: CanvasElement) {
+      const foundElement = state.canvasElements.find((canvasElement: CanvasElement) => canvasElement.id === payload.id)
+      if (foundElement) {
+        foundElement.isVisible = true
```

Comment:
The canvasElements have a isVisible field, telling the layer if it should be rendered or destroyed.